lib/instance_billing_flavor_check/utils.py

```diff
@@ -0,0 +1,179 @@
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+        logging.FileHandler('instance-billing-flavor-check.log', mode='w'),
+    ],
+    level=logging.INFO,
+    format="%(message)s"
+)
+try:
+    from cloudregister.registerutils import get_smt
+except ModuleNotFoundError as err:
+    pass
+
+REGION_SRV_CLIENT_CONFIG_PATH = '/etc/regionserverclnt.cfg'
+ETC_OS_RELEASE_PATH = '/etc/os-release'
+ETC_HOSTS_PATH = '/etc/hosts'
+
+requests.packages.urllib3.disable_warnings(
+    requests.packages.urllib3.exceptions.InsecureRequestWarning
+)
+
+ 
+def get_instance_data_command():
+    config = configparser.ConfigParser()
+    if config.read(REGION_SRV_CLIENT_CONFIG_PATH):
+        try:
+            return config['instance']['dataProvider']
+        except Exception as err:
+            logger.error(
+                "Could not parse %s: %s", REGION_SRV_CLIENT_CONFIG_PATH, err
+            )
+    else:
+        logger.error("Could not read file %s", REGION_SRV_CLIENT_CONFIG_PATH)
+
+
+def get_metadata():
+    """Return instance metadata."""
+    command = get_instance_data_command()
+    if not command:
+        return
+
+    command = command.split(' ')
+    result = Command.run(command)
+    if result.returncode == 0:
+        return result.output
+
+    logger.error(
+        "Could not fetch the metadata after running the command '%s': '%s' with status '%s'",
+        ' '.join(command),
+        result.error,
+        result.returncode
+    )
+
+
+def get_identifier():
+    """Return the identifier found in /etc/os-release."""
+    try:
+        with open(ETC_OS_RELEASE_PATH, encoding='utf-8') as stream:
+            csv_reader = csv.reader(stream, delimiter="=")
+            os_release = dict(csv_reader)
+        return os_release.get('NAME')
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_OS_RELEASE_PATH)
+
+
+def get_rmt_ip_addr():
+    """Return the RMT update server IP the instance is registered to."""
+    try:
+        with open(ETC_HOSTS_PATH, encoding='utf-8') as etc_hosts:
+            etc_hosts_lines = etc_hosts.readlines()
+    except FileNotFoundError:
+        logger.error("Could not open '%s' file", ETC_HOSTS_PATH)
+        return
+
+    # use present RMT server IP first if registered
+    for etc_hosts_line in etc_hosts_lines:
+        if 'susecloud.net' in etc_hosts_line:
+            return etc_hosts_line.split('\t')[0]
+
+    if 'cloudregister' in sys.modules:
+        # get a new RMT IP
+        server = get_smt(False)
+        rmt_ips_addr = []
+        if server.get_ipv6():
+            rmt_ips_addr.append('[{}]'.format(server.get_ipv6()))
+        rmt_ips_addr.append(server.get_ipv4())
+        return rmt_ips_addr
+
+    logger.info('Could not determine update server IP address')
+
+
+def make_request(rmt_ip_addr, metadata, identifier):
+    """Return the flavour from the RMT server request."""
+    instance_check_url = f"https://{rmt_ip_addr}/api/instance/check"
```

Comment:
The code also has to work with SLE 12, i.e. Python 3.4 format strings do not exist there yet, you have to use `.format`

Reply:
Yes, I was not sure how old the versions we are supporing would be (maybe from SLES12 SP5, maybe older)
So in [e88dd7a](https://github.com/SUSE-Enceladus/instance-billing-flavor-check/pull/1/commits/e88dd7a2e3b783e9e8520a47f0a70d95a59360c1) I started using `format`
and yes, will update the rest of string formatting